### PR TITLE
Expose ephemeral-storage in resource config

### LINKED ISSUE #1390: expose ephemeral storage configuration in resources
## Feature Request

Ref: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#local-ephemeral-storage

related issue: #1386

### LINKED ISSUE #1395: Feature request: Add ephemeral storage for TiDB pod
## Feature Request

Currently, TiDB does not use storage, the slow log and audit log are using temporary storage. TiDB will soon support disk based computation to avoid large dataset consuming too much memory causing OOM. https://github.com/pingcap/tidb/issues/11607

We should consider adding ephemeral storage for TiDB pod.

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

close #1390 
close #1395 

Tests 

 - Unit test
 - E2E test

Some UT cases are removed because the type constraint makes the quantity format error will never happen in the Go code, instead, we will encounter an error when post a wrong request body to kube-apiserver, which could be verified in e2e or stability test.
### Does this PR introduce a user-facing change?:
 
 ```release-note
expose ephemeral-storage in resource config
 ```

@cofyc @weekface @tennix @Yisaer PTAL

### CODE REVIEW

File: pkg/tkctl/cmd/info/info.go [hunk 1]
```diff
@@ -153,6 +154,48 @@ func (o *InfoOptions) Run() error {
...
+	var tidbStorage resource.Quantity
+	if tc.Spec.TiDB.Requests != nil {
+		if cpu := tc.Spec.TiDB.Requests.Cpu(); cpu != nil {
+			tidbCPU = *cpu
+		}
+		if mem := tc.Spec.TiDB.Requests.Memory(); mem != nil {
+			tidbMemory = *mem
+		}
+		if st, ok := tc.Spec.TiDB.Requests[v1.ResourceStorage]; ok {
+			tidbMemory = st
```
Comment (weekface) [inline pkg/tkctl/cmd/info/info.go]:
```suggestion
			tidbStorage = st
```

**Comment by aylei** [comment]
I solved a merge conflict, @weekface @cofyc could you please take a look again?

**Pull request merged**